### ISSUE TITLE
wasienv and wasi-sdk to build webassembly packages (wasi)

### ISSUE THREAD

**Issue by rambip** [NixOS/nixpkgs#182356]
**Project description**
    
Wasienv is a tool that aims to bring all programming languages to [WebAssembly WASI](https://github.com/WebAssembly/WASI). 

With wasienv you can compile:

    - C/C++ projects to Wasm + WASI ([see usage example](https://docs.wasmer.io/ecosystem/wasienv/compile-c-c++-to-wasm-wasi))
    - Swift to Wasm + WASI ([see usage example](https://docs.wasmer.io/ecosystem/wasienv/compile-swift-to-wasm-wasi))

**Metadata**

* homepage URL: https://github.com/wasienv/wasienv
* source URL: https://github.com/wasienv/wasienv
* license: mit
* platforms: unix, linux, darwin

**Comment by rvolosatovs** [comment]
It's been 1.5 years since last update to that repo, what's the maintenance status of it?

**Comment by rambip** [comment]
Right, I didn't realize it was basically a wrapper around [wasi-sdk](https://github.com/WebAssembly/wasi-sdk)

This one is maintained, the last change was 2 months ago.

I have a nix-expression for it, but I don't have the computing ressources necessary to compile it from source:

```nix
wasi_sdk_src = pkgs.stdenv.mkDerivation {
    name = "wasi-sdk";
    version = "16.0";
    dontConfigure = true;
    buildInputs = [pkgs.clang pkgs.cmake pkgs.ninja pkgs.python3];
    src = pkgs.fetchFromGitHub {
        owner = "WebAssembly";
        repo = "wasi-sdk";
        rev = "wasi-sdk-16";
        hash = "sha256-inHa5pps15dHT2BPD/WEoXIJ7njDlGXObx7tfZOi1/I";
        fetchSubmodules = true;
    };
    buildPhase = ''
    NINJA_FLAGS=-v make package
    '';
    installPhase = ''
    mkdir -p $out
    cp -r dist $out/bin
    '';
};
```

Should I edit the issue's title ?

**Comment by rvolosatovs** [comment]
Sure, although I am not sure `wasi-sdk` is required either, since AFAIK it, in itself, is a wrapper around https://github.com/WebAssembly/wasi-libc, which is already packaged.

You can directly compile C for WASI, see some examples here https://github.com/enarx/codex/blob/390c4ca47d1f30de09f92def6e1ec27cb93c5726/flake.nix#L71-L80
